File: services/assessment-service/src/main/java/com/visor/school/assessment/repository/AssessmentRepository.java
package com.visor.school.assessment.repository;

import com.visor.school.assessment.model.Assessment;
import com.visor.school.assessment.model.AssessmentStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface AssessmentRepository extends JpaRepository<Assessment, UUID> {
    List<Assessment> findByClassId(UUID classId);
    List<Assessment> findByStatus(AssessmentStatus status);
    List<Assessment> findByClassIdAndStatus(UUID classId, AssessmentStatus status);
}

File: services/academic-assessment-service/src/main/kotlin/com/visor/school/assessment/repository/AssessmentRepository.kt
package com.visor.school.assessment.repository

import com.visor.school.assessment.model.Assessment
import com.visor.school.assessment.model.AssessmentStatus
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface AssessmentRepository : JpaRepository<Assessment, UUID> {
    fun findByClassId(classId: UUID): List<Assessment>
    fun findByStatus(status: AssessmentStatus): List<Assessment>
    fun findByClassIdAndStatus(classId: UUID, status: AssessmentStatus): List<Assessment>
}

